### PR TITLE
Fix ChronicleMethodReaderTest.shouldFilterByMultipleInclusionRegex

### PR DESCRIPTION
**Pull request by yevgenp**
Closes https://github.com/OpenHFT/Chronicle-Queue/issues/1150, https://github.com/OpenHFT/Chronicle-Queue/issues/821

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java [hunk 1]
```diff
@@ -187,10 +187,12 @@ public void shouldFilterByInclusionRegex() {
...
-    @Ignore("https://github.com/OpenHFT/Chronicle-Queue/issues/1150")
     @Test
     public void shouldFilterByMultipleInclusionRegex() {
-        basicReader().withInclusionRegex(".*bye$").withInclusionRegex(".*o.*").execute();
+        basicReader()
+                .withInclusionRegex(".*bye.*")
+                .withInclusionRegex(".*o.*")
+                .execute();
 
         assertEquals(24, capturedOutput.size());
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java:197]: is 24 correct?

File: src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java [hunk 2]
```diff
@@ -187,10 +187,12 @@ public void shouldFilterByInclusionRegex() {
                 .forEach(msg -> assertThat(msg, containsString("goodbye")));
     }
 
-    @Ignore("https://github.com/OpenHFT/Chronicle-Queue/issues/1150")
     @Test
     public void shouldFilterByMultipleInclusionRegex() {
-        basicReader().withInclusionRegex(".*bye$").withInclusionRegex(".*o.*").execute();
+        basicReader()
+                .withInclusionRegex(".*bye.*")
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java:193]: @yevgenp does this include OR or AND? We should specify in the javadoc what happens if you specify multiple inclusion regexes.

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java [hunk 3]
```diff
@@ -586,4 +586,72 @@ public void run() throws InterruptedException {
...
+                assertTrue(tailer.toString().contains("StoreTailer{"));
+                tailer.toStart();//forbidden
+            }
+        }
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void cantMoveToEndDuringDocumentReading() {
+        File dir = getTmpDir();
+        try (SingleChronicleQueue queue = ChronicleQueue.singleBuilder(dir).build();
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java]: as a general note, we should use `testBlockSize` in tests as per https://github.com/OpenHFT/Chronicle-Queue/commit/91bc0e9aac2f15f9d11bd73d9ae73ea91f41a3f2

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java [hunk 4]
```diff
@@ -586,4 +586,72 @@ public void run() throws InterruptedException {
...
+             ExcerptTailer tailer = queue.createTailer();
+             ExcerptAppender appender = queue.createAppender()) {
+            appender.writeText("Hello World");
+            try (DocumentContext dc = tailer.readingDocument(true)) {
+                assertTrue(dc.isPresent());
+                assertTrue(dc.isMetaData());
+                assertEquals("header", dc.wire().readEvent(String.class));
+                assertTrue(tailer.toString().contains("StoreTailer{"));
+                tailer.toStart();//forbidden
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java:603]: love the additional test coverage

File: src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java (hunk 1 above)
Comment (yevgenp) [reply src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java:197]: Yes

File: src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java (hunk 2 above)
Comment (yevgenp) [reply src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java:193]: AND

File: src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java (hunk 3 above)
Comment (yevgenp) [reply src/test/java/net/openhft/chronicle/queue/impl/single/StoreTailerTest.java]: Added

File: src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java (hunk 2 above)
Comment (JerryShea) [reply src/test/java/net/openhft/chronicle/queue/internal/reader/ChronicleMethodReaderTest.java:193]: @tgd I have rebased on develop today. Please merge if you are happy with this

**Review by JerryShea — APPROVED** [review]

**Review by tgd — APPROVED** [review]

**Pull request merged**